### PR TITLE
Add delete button to project modal

### PR DESCRIPTION
**Pull request by ema-fox**

### CODE REVIEW

File: src/sb/app/project/data.cljc [hunk 1]
```diff
@@ -128,4 +128,12 @@
                                                            #{:role/project-admin})]
     (validate/assert project :project/as-map)
     (db/transact! [membership])
-    {:entity/id (:entity/id project)}))
\ No newline at end of file
+    {:entity/id (:entity/id project)}))
+
+(q/defx delete!
+  "Mutation fn. Marks project as deleted by given project-id."
+  {:endpoint {:post "/p/:project-id/delete"}}
```
Comment (mhuebert) [inline src/sb/app/project/data.cljc:135]: Could we verify if this metadata is still necessary? I wonder if it was left over in the org delete endpoint (my fault), but no longer required or used.

File: src/sb/app/project/ui.cljc [hunk 2]
```diff
@@ -126,6 +127,23 @@
      [entity.ui/persisted-attr project :entity/admission-policy props])]
   )
 
+;; TODO: taken from board.ui, merge into common component if sensible
```
Comment (mhuebert) [inline src/sb/app/project/ui.cljc:130]: Agree, let's move this to a shared namespace as it's a very general implementation

File: src/sb/app/project/data.cljc (hunk 1 above)
Comment (ema-fox) [reply src/sb/app/project/data.cljc:135]: Just confirmed that it is not necessary.

**Comment by mhuebert** [comment]
Do we know if there are queries that need to be updated to not show deleted projects?

**Review by mhuebert — APPROVED** [review]

**Pull request merged**